anchore_engine/services/policy_engine/engine/policy/gate_util_provider.py

```diff
@@ -119,9 +118,19 @@ def oldest_namespace_feed_sync(
                 return None
 
     def have_vulnerabilities_for(self, distro_namespace_obj: DistroNamespace) -> bool:
-        wrapper = GrypeWrapperSingleton.get_instance()
-        source_counts = wrapper.query_record_source_counts()
-        groups = [source.group for source in source_counts if source.count > 0]
+        with session_scope() as session:
+            try:
+                grypedb = get_most_recent_active_grypedb(session)
+            except NoActiveGrypeDB:
+                logger.info(
+                    "No vulnerabilities for image distro found because no active grype db found"
+                )
+                return False
+
+            groups = [
+                group["name"] for group in grypedb.groups if group["record_count"] > 0
```

Comment:
Can grypedb.groups be an invalid reference? Might be worth a check before accessing it

Reply:
@nightfurys  Do you mean in the case that grypedb is potentially None? That should be handled by `get_most_recent_active_grypedb` which raises NoActiveGrypeDB if the query does not return any results. I can add that check for safety here as well though, but want to confirm that is what you mean by invalid reference.

Reply:
2 cases I can see: grypedb = None, or grypdb.groups = None.

Reply:
I think yes, grypedb=None or grypedb.groups=None. If those are handled in the contract of that function call that should cover it.

Reply:
Oh i just pushed the changes to add that right before I saw this response @zhill

Reply:
`get_most_recent_active_grypedb` does not check if groups is None so the addition of that check here in this function is a good one

Reply:
@zburstein I meant grypedb.groups != None should be checked. Was that check been added?